portal-impl/src/com/liferay/portal/service/impl/RegionLocalServiceImpl.java

```
@@ -68,78 +68,107 @@ public void deleteCountryRegions(long countryId) {
 	}
 
 	@Override
 	public Region fetchRegion(long countryId, String regionCode) {
-		return null;
+		return regionPersistence.fetchByC_R(countryId, regionCode);
 	}
 
 	@Override
-	public Region getRegion(long countryId, String regionCode) {
-		return null;
+	public Region getRegion(long countryId, String regionCode)
+		throws PortalException {
+
+		return regionPersistence.findByC_R(countryId, regionCode);
 	}
 
 	@Override
-	public List<Region> getRegions(long countryId, boolean active) {
-		return null;
+	public List<Region> getRegions(long countryId, boolean active)
+		throws PortalException {
+
+		return regionPersistence.findByC_A(countryId, active);
 	}
 
 	@Override
 	public List<Region> getRegions(
 		long countryId, boolean active, int start, int end,
 		OrderByComparator<Region> orderByComparator) {
 
-		return null;
+		return regionPersistence.findByC_A(
+			countryId, active, start, end, orderByComparator);
 	}
 
 	@Override
 	public List<Region> getRegions(
 		long countryId, int start, int end,
 		OrderByComparator<Region> orderByComparator) {
 
-		return null;
+		return regionPersistence.findByCountryId(
+			countryId, start, end, orderByComparator);
 	}
 
 	@Override
-	public List<Region> getRegions(long companyId, String a2, boolean active) {
-		return null;
+	public List<Region> getRegions(long companyId, String a2, boolean active)
+		throws PortalException {
+
+		Country country = countryPersistence.findByC_A2(companyId, a2);
+
+		return regionPersistence.findByC_A(country.getCompanyId(), active);
 	}
 
 	@Override
 	public int getRegionsCount(long countryId) {
-		return null;
+		return regionPersistence.countByCountryId(countryId);
 	}
 
 	@Override
 	public int getRegionsCount(long countryId, boolean active) {
-		return null;
+		return regionPersistence.countByC_A(countryId, active);
 	}
 
 	@Override
 	public Region updateActive(long regionId, boolean active) {
-		return null;
+		Region region = regionPersistence.fetchByPrimaryKey(regionId);
```

This should use `findByPrimaryKey` to throw an exception when the regionId is invalid. Otherwise, it's possible for NPE to occur on the next line.